src/main/kotlin/io/kotest/plugin/intellij/structure/KotestStructureViewExtension.kt

```diff
@@ -24,25 +24,48 @@ import org.jetbrains.kotlin.psi.KtClassOrObject
  */
 class KotestStructureViewExtension : StructureViewExtension {
 
+   private var cachedParent: PsiElement? = null
```

Comment:
Since you're updating this from a background thread, I think you should use something like `AtomicReference` or `volatile`, no?